amazon_cloud_code_generator/data/plugins/module_utils/core.py

```diff
@@ -325,6 +325,10 @@ def check_in_progress_requests(self, type_name: str, identifier: str):
     def wait_for_in_progress_requests(
         self, in_progress_requests: List, identifier: str
     ):
+        # Dont warn if nothing to wait on
+        if not in_progress_requests:
```

Comment:
It’s a nitpick. You can also add the of check and enclose everything inside without adding using return.